File: production/client/src/main/scala/kpn/client/components/network/routes/NetworkRouteFilterCriteria.scala
// TODO migrate to Angular
package kpn.client.components.network.routes

import kpn.client.filter.TimeFilterKind

case class NetworkRouteFilterCriteria(
  page: Int = 1,
  investigate: Option[Boolean] = None,
  accessible: Option[Boolean] = None,
  roleConnection: Option[Boolean] = None,
  lastUpdated: TimeFilterKind.Value = TimeFilterKind.ALL
)

File: client/src/app/analysis/pages/network/routes/network-route-filter-criteria.ts
import {TimestampFilterKind} from "../../../../kpn/filter/timestamp-filter-kind";

export class NetworkRouteFilterCriteria {

  constructor(readonly investigate: boolean = null,
              readonly accessible: boolean = null,
              readonly roleConnection: boolean = null,
              readonly relationLastUpdated: TimestampFilterKind = TimestampFilterKind.ALL) {
  }
}
